presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/json/JsonBasedUdfFunctionMetadata.java

```diff
@@ -50,27 +59,45 @@
      */
     private final RoutineCharacteristics routineCharacteristics;
 
+    /**
+     * Optional Aggregate-specific metadata (required for aggregation functions)
+     */
+    private final Optional<AggregationFunctionMetadata> aggregateMetadata;
+
     @JsonCreator
     public JsonBasedUdfFunctionMetadata(
             @JsonProperty("docString") String docString,
-            @JsonProperty("outputType") String outputType,
-            @JsonProperty("paramTypes") List<String> paramTypes,
+            @JsonProperty("functionKind") FunctionKind functionKind,
+            @JsonProperty("outputType") TypeSignature outputType,
+            @JsonProperty("paramTypes") List<TypeSignature> paramTypes,
             @JsonProperty("schema") String schema,
-            @JsonProperty("routineCharacteristics") RoutineCharacteristics routineCharacteristics)
+            @JsonProperty("routineCharacteristics") RoutineCharacteristics routineCharacteristics,
+            @JsonProperty("aggregateMetadata") Optional<AggregationFunctionMetadata> aggregateMetadata)
     {
         this.docString = requireNonNull(docString, "docString is null");
+        this.functionKind = requireNonNull(functionKind, "functionKind is null");
         this.outputType = requireNonNull(outputType, "outputType is null");
         this.paramTypes = ImmutableList.copyOf(requireNonNull(paramTypes, "paramTypes is null"));
         this.schema = requireNonNull(schema, "schema is null");
         this.routineCharacteristics = requireNonNull(routineCharacteristics, "routineCharacteristics is null");
+
+        if (functionKind == AGGREGATE && (aggregateMetadata == null || !aggregateMetadata.isPresent())) {
+            throw new IllegalStateException("Aggregate Functions require aggregate metadata");
+        }
+        this.aggregateMetadata = aggregateMetadata;
```

Comment:
If aggregateMetadata is a nullable non-optional field, we should do 
`this.aggregateMetadata = Optional.fromNullable(aggregateMetadata)`.  
or if it's not nullable, but optional, we should add a requireNonNull check here

Reply:
Will add a `requireNonNull`. Thanks.